syntax = "proto3";
package sqlpkgs;




import "database_delivery_sdk/model/database_delivery/sql_package_version.proto";
import "database_delivery_sdk/model/database_delivery/app.proto";
import "database_delivery_sdk/model/database_delivery/dbservice.proto";

/*
ListSQLPackage请求
*/
message ListSQLPackageRequest {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页数据量
    */
    int32 pageSize = 2 ;
    /*
    指定查询条件,应用 id
    */
    string appId = 3 ;
    /*
    指定查询条件,数据库服务 id
    */
    string dbServiceId = 4 ;
}

/*
ListSQLPackage返回
*/
message ListSQLPackageResponse {
message List {
    /*
    SQL包下的版本列表
    */
    repeated database_delivery.SQLPackageVersion versionList = 1 ;
    /*
    SQL包关联的应用列表
    */
    repeated database_delivery.Application APP = 2 ;
    /*
    SQL包关联的数据库服务列表
    */
    repeated database_delivery.DBService DBSERVICE = 3 ;
    /*
    SQL包ID
    */
    string id = 4 ;
    /*
    名称
    */
    string name = 5 ;
    /*
    描述
    */
    string memo = 6 ;
    /*
    创建者
    */
    string creator = 7 ;
    /*
    ctime
    */
    int64 ctime = 8 ;
    /*
    mtime
    */
    int64 mtime = 9 ;
    /*
    文件仓库包ID
    */
    string repoPackageId = 10 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListSQLPackageApi返回
*/
message ListSQLPackageResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    sqlpkgs.ListSQLPackageResponse data = 4 ;
}

